src/git_config/git_config_entry.rs

```diff
@@ -32,19 +46,45 @@ lazy_static! {
         "
     )
     .unwrap();
+    static ref GITLAB_REMOTE_URL: Regex = Regex::new(
+        r"(?x)
+        ^
+        (?:https://|git@)? # Support both HTTPS and SSH URLs, SSH URLs optionally omitting the git@
+        gitlab\.com
+        [:/]              # This separator differs between SSH and HTTPS URLs
+        ([^/]+)           # Capture the user/org name
+        (/.*)?            # Capture group(s), if any
+        /
+        (.+?)             # Capture the repo name (lazy to avoid consuming '.git' if present)
+        (?:\.git)?        # Non-capturing group to consume '.git' if present
+        $
+        "
+    )
+    .unwrap();
 }
 
 impl FromStr for GitRemoteRepo {
     type Err = Error;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         if let Some(caps) = GITHUB_REMOTE_URL.captures(s) {
-            Ok(Self::GitHubRepo(format!(
-                "{user}/{repo}",
-                user = caps.get(1).unwrap().as_str(),
-                repo = caps.get(2).unwrap().as_str()
-            )))
+            Ok(Self::GitHubRepo {
+                repo_slug: format!(
+                    "{user}/{repo}",
+                    user = caps.get(1).unwrap().as_str(),
+                    repo = caps.get(2).unwrap().as_str()
+                ),
+            })
+        } else if let Some(caps) = GITLAB_REMOTE_URL.captures(s) {
+            Ok(Self::GitLabRepo {
+                repo_slug: format!(
+                    "{user}{groups}/{repo}",
+                    user = caps.get(1).unwrap().as_str(),
+                    groups = caps.get(2).map(|x| x.as_str()).unwrap_or_default(),
+                    repo = caps.get(3).unwrap().as_str()
+                ),
+            })
```

Comment:
If you felt like it, it might be nice to separate these into separate functions and have `from_str` implemented as "return the result from the first parser function that succeeds".

Reply:
My first thought, before the weekend, was to see if it could be done using serde.. But that seemed like total overkill 😅

It would be nice with something else in the long run.. Like, each repo type just knowing how its URL is represented, then registering itself so that the parser can try them all.

That could enable specifying full-fledged new repo types in a config file, say, for those who have hosted GitHub or GitLab instances, etc.